src/boot.c

```diff
@@ -478,23 +480,34 @@ void read_boot(DOS_FS * fs)
     if (verbose)
 	printf("Checking we can access the last sector of the filesystem\n");
     /* Can't access last odd sector anyway, so round down */
-    fs_test((off_t)((total_sectors & ~1) - 1) * logical_sector_size,
-	    logical_sector_size);
+    position = (long long)((total_sectors & ~1) - 1) * logical_sector_size;
+    if (position != (off_t)position)
```

Comment:
I'm thinking about defining `OFF_MAX` value as:
```c
#include <limits.h>
#ifndef OFF_MAX
#define OFF_MAX ((1ULL << (sizeof(off_t) * CHAR_BIT - 1)) - 1)
#endif
```
And then comparing `position` against it. But I'm not sure if it is a good idea...

Reply:
@stoeckmann do you have any opinion if above is a good idea or not?

Reply:
Sorry for not getting back at this in a timely manner. I've already tested it just to see if it works correctly and it does.

I'm fine with that approach. It's just that I tried to avoid implementing it like that for assuming that there is a good reason that OFF_MAX is not part of limits.h.

Reply:
Looks like that some systems provides `OFF_MAX` macro in limits.h (e.g. freebsd). Some software defines own `OFF_T_MAX` macro.

Reply:
Cool, that makes it even more obvious to be a good solution. I'll adjust my pull request.

Reply:
Anyway, in pull request https://github.com/dosfstools/dosfstools/pull/84#pullrequestreview-202281703 is exactly same problem, checking if value in long long can be stored into off_t.

So we should choose one solution for this `off_t` problem and use it on both places.

So maybe `OFF_MAX` should be defined in some header file?

Reply:
In that case it should be done in an own refactoring commit. At first commons.h looked like a good spot, but that file itself doesn't even contain off_t. Adding off_t to fsck.fat.h would have made sense, because that file uses off_t in the offending fashion. But then again it would have to be repeated at least twice as well.

After seeing the definition of TRUE and FALSE in mkfs.fat.c I decided to not play around with OFF_MAX much further, because these adjustments better fit into own commits.

Reply:
`src/common.h` is included in all source files, so seems as the best place for defining OFF_MAX. It is not a problem that file does not have any function with `off_t` type. So please put it there. Defining OFF_MAX on two places does make sense.

TRUE/FALSE macros are just some historic code and candidate for future removal...

Reply:
Added to src/common.h as requested.